tests/Oryx.BuildImage.Tests/ExtensibilityModelBuildScriptTest.cs

```diff
@@ -0,0 +1,316 @@
+                    {
+                        Assert.True(result.IsSuccess);
+                        Assert.Contains(
+                            $"echo \"Hello, world!\"",
+                            result.StdOut);
+                        Assert.Contains(
+                            $"echo \"Hello, foobar!\"",
+                            result.StdOut);
+                    },
+                    result.GetDebugInfo());
+            }
+            finally
+            {
+                DeleteTempDirectory();
+            }
+        }
+
+        [Fact, Trait("category", "githubactions")]
+        public void BuildScriptGenerated_WithExtensibilityModel_WithHttpGet_Succeeds()
+        {
+            // Arrange
+            var configContent = @"
+base-os: debian
+pre-build: 
+  - description: 'Test description'
+    http-get:
+      url: 'testurl'
+      file-name: 'samplefile'
+      headers:
+        - 'header1'
+";
+
+            var volume = CreateWebFrontEndVolume();
+            CreateTempDirectory();
+            try
+            {
+                CreateExtensibilityConfigFile(configContent);
+                var tempDirVolume = CreateTempDirectoryVolume();
+                var appDir = volume.ContainerDir;
+                var tempDir = tempDirVolume.ContainerDir;
+                var script = new ShellScriptBuilder()
+                    .AddCommand($"cp {tempDir}/{FilePaths.ExtensibleConfigurationFileName} {appDir}")
+                    .AddCommand($"oryx build-script {appDir}")
+                    .ToString();
+
+                // Act
+                var result = _dockerCli.Run(new DockerRunArguments
+                {
+                    ImageId = _imageHelper.GetGitHubActionsBuildImage(),
+                    Volumes = new List<DockerVolume> { volume, tempDirVolume },
+                    CommandToExecuteOnRun = "/bin/bash",
+                    CommandArguments = new[] { "-c", script }
+                });
+
+                // Assert
+                RunAsserts(
+                    () =>
+                    {
+                        Assert.True(result.IsSuccess);
+                        Assert.Contains(
+                            $"curl -o samplefile -H header1 testurl",
+                            result.StdOut);
+                    },
+                    result.GetDebugInfo());
+            }
+            finally
+            {
+                DeleteTempDirectory();
+            }
+        }
+
+        [Fact, Trait("category", "githubactions")]
+        public void BuildScriptGenerated_WithExtensibilityModel_WithMultipleProperties_Succeeds()
+        {
+            // Arrange
+            var configContent = @"
+base-os: debian
+env: 
+  - name: FOO
+    value: BAR
+pre-build: 
+  - description: 'Test description'
+    scripts:
+      - 'echo ""Hello, world!""'
+      - 'echo ""Hello, foobar!""'
+    http-get:
+      url: 'testurl'
+      file-name: 'samplefile'
+      headers:
+        - 'header1'
```

Comment:
maybe add another header here to test that case? Also, don't we support multiple pre-build commands, like 
```
base-os: debian
env: 
  - name: FOO
    value: BAR
pre-build: 
  - description: 'Test description'
    scripts:
      - 'echo ""Hello, world!""'
      - 'echo ""Hello, foobar!""'
    http-get:
      url: 'testurl'
      file-name: 'samplefile'
      headers:
        - 'header1'
        - 'header2'
  - description: 'Test description 2'
    scripts:
      - 'echo ""Hello, lorem!""'
      - 'echo ""Hello, ipsum!""'
    http-get:
      url: 'testurl2'
      file-name: 'samplefile2'
      headers:
        - 'header3'
        - 'header4'
```

Reply:
Good idea, I'll add a test to handle multiple headers and a test to handle multiple pre-build commands.